### PR TITLE
Brings new `plugins` command output message as per #3212

### LINKED ISSUE #3212: Improve output of `plugins`
The output of `plugins` should improve and be more readable.
  
This is how it looks like now:
```
In file:/data/rw/code/scala/sbt/
	sbt.plugins.IvyPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	sbt.plugins.JvmPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	sbt.plugins.CorePlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	sbt.plugins.JUnitXmlReportPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	sbt.plugins.Giter8TemplatePlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	com.lucidchart.sbt.scalafmt.ScalafmtCorePlugin: enabled in sbtRoot
	com.lucidchart.sbt.scalafmt.ScalafmtPlugin: enabled in sbtRoot
	bintray.BintrayPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	sbt.contraband.ContrabandPlugin: enabled in testingProj, protocolProj, runProj, mainProj, commandProj
	sbt.contraband.JsonCodecPlugin: enabled in testingProj, protocolProj, commandProj
	PublishBinPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	ScriptedPlugin: enabled in sbtRoot
	NightlyPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
	SbtLauncherPlugin: enabled in bundledLauncherProj
	StatusPlugin: enabled in testingProj, sbtRoot, protocolProj, scriptedSbtProj, stdTaskProj, sbtProj, taskProj, runProj, bundledLauncherProj, mainProj, scriptedPluginProj, coreMacrosProj, mainSettingsProj, actionsProj, commandProj, testAgentProj
```
  
Ideally, we would like to show clearly the following items:
  
* A header for every plugin (with its fully qualified name)
* Indented by two spaces, a list of all the projects in which the plugin is enabled on. (Experiment with listing the projects each one in a new line.
* Use colors for the plugin name, to make the output more readable.
  
Another approach is to show all the plugins enabled per project, instead of all projects in which a plugin is enabled.
  
I feel this is a perfect candidate for the [Scala Center spree](https://github.com/scalacenter/sprees) in Scaladays Copenhague.

### PR DESCRIPTION
**Pull request by aaabramov**
Fixes https://github.com/sbt/sbt/issues/3212

Previous output:
image

New output:
image

### CODE REVIEW

File: main/src/main/scala/sbt/internal/PluginsDebug.scala [hunk 1]
```diff
@@ -121,17 +121,33 @@ private[sbt] object PluginsDebug {
...
-            build.projects.toList.filter(_.autoPlugins.contains(plugin)).map(_.id)
-          val actString =
-            if (activatedIn.nonEmpty) activatedIn.mkString(": enabled in ", ", ", "")
-            else "" // TODO: deal with large builds
-          s"\n\t${plugin.label}$actString"
-        }
-        s"In $uri${pluginStrings.mkString}"
-      }
+
+      def helpBuild(uri: URI, build: LoadedBuildUnit): String =
```
Comment (aaabramov) [inline main/src/main/scala/sbt/internal/PluginsDebug.scala]: The build `uri` parameter is not used here. Should we start output with a line like
```
In build ${uri}:
  Enabled plugins in module1 (/Users/aabramov/dev/open-source/sbt-3212/module-1):
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
  Enabled plugins in module2 (/Users/aabramov/dev/open-source/sbt-3212/module-2):
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
  Enabled plugins in module3 (/Users/aabramov/dev/open-source/sbt-3212/module-3):
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
```
?

Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/PluginsDebug.scala]: Maybe it's better to print the full absolute path once at the top as build URI, and then skip the `(/Users/aabramov/dev/open-source/sbt-3212/module-1)`?

**Comment by eed3si9n** [comment]
@aaabramov Thanks for this. Looks like a great improvement forward.

File: main/src/main/scala/sbt/internal/PluginsDebug.scala (hunk 1 above)
Comment (aaabramov) [reply main/src/main/scala/sbt/internal/PluginsDebug.scala]: Agree.
So let it be
```
In build /Users/aabramov/dev/open-source/sbt-3212:
  Enabled plugins in module1:
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
  Enabled plugins in module2:
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
  Enabled plugins in module3:
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
    ...
```
, am I right?

Comment (aaabramov) [reply main/src/main/scala/sbt/internal/PluginsDebug.scala]: Something like this:
image

**Comment by aaabramov** [comment]
And another question. What should we do if there are no enabled plugins? I believe it's rare situation because sbt adds some plugins by default.

Should we just skip the listing, like
```
In build /Users/aabramov/dev/open-source/sbt-3212/:
  Enabled plugin in module1:
  Enabled plugins in module2:
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
```
or add notice?
```
In build /Users/aabramov/dev/open-source/sbt-3212/:
  Enabled plugins in module1:
    <none>
  Enabled plugins in module2:
    sbt.plugins.CorePlugin
    sbt.plugins.Giter8TemplatePlugin
```

**Comment by aaabramov** [comment]
@eed3si9n shell we merge this?

**Comment by eed3si9n** [comment]
I realized that this is targeting 1.3.x branch. Given that 1.4.0 is out, this would go into either 1.4.1 or 1.5.0.
Either case, could you cherry pick this commit on top of `develop` branch?

This is a breaking change, but I think it's safe enough so I'm ok with this going with 1.4.1. @eatkins WDYT?

**Comment by aaabramov** [comment]
I am fine with pointing it to `develop`. Working on it

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**